src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -1366,6 +1367,67 @@ bool FancyZones::ShouldProcessSnapHotkey(DWORD vkCode) noexcept
     return false;
 }
 
+void FancyZones::ApplyQuickLayout(int key) noexcept
+{
+    std::unique_lock writeLock(m_lock);
+
+    HMONITOR monitor;
+    if (m_settings->GetSettings()->spanZonesAcrossMonitors)
+    {
+        monitor = NULL;
+    }
+    else
+    {
+        POINT cursorPoint;
+        if (!GetCursorPos(&cursorPoint))
+        {
+            return;
+        }
+
+        monitor = MonitorFromPoint(cursorPoint, MONITOR_DEFAULTTOPRIMARY);
+    }

```

Comment:
I guess there might be other places where we need a similar logic, can we move it to its own helper?
`ApplyQuickLayout` should not know that `spanZonesAcrossMonitors` even exists and in the future we might support spanning only some monitor and not all monitors, so this logic should be placed in a dedicated helper.

Reply:
Makes sense